### PR TITLE
Bastard Keyboards: Add support for Dilemma Max (4x6_4)

### PR DESCRIPTION
**Pull request by casuanoob**
## Description

This PR adds support for @bstiq's Dilemma Max keyboard, which is a 4x6+4 addition to the Dilemma range of keyboards.
- Add implementation for the `dilemma_4x6_4` keyboard
- Add basic `default` keymap.
- Add `via` keymap.
- Update `bastardkb/dilemma/readme.md`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bastardkb/dilemma/4x6_4/config.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#    define SPLIT_TRANSPORT_MIRROR
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED
+#    define RGB_MATRIX_KEYPRESSES
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+
+// Startup values.
+#    define RGB_MATRIX_DEFAULT_HUE 0
+#    define RGB_MATRIX_DEFAULT_SAT 255
+#    define RGB_MATRIX_DEFAULT_VAL 64
+#    define RGB_MATRIX_DEFAULT_HSV RGB_MATRIX_DEFAULT_HUE, RGB_MATRIX_DEFAULT_SAT, RGB_MATRIX_DEFAULT_VAL
```
Comment (zvecr) [inline keyboards/bastardkb/dilemma/4x6_4/config.h:57]:
```suggestion
```

**Comment by casuanoob** [comment]
Thanks for the review! I've applied the requested changes.

File: keyboards/bastardkb/dilemma/4x6_4/config.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#    define RGB_MATRIX_DEFAULT_SAT 255
+#    define RGB_MATRIX_DEFAULT_VAL 64
+
+// Rainbow swirl as startup mode.
+#    define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
+#    define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_LEFT_RIGHT
+
+// Slow swirl at startup.
+#    define RGB_MATRIX_DEFAULT_SPD 32
+#endif
```
Comment (drashna) [inline keyboards/bastardkb/dilemma/4x6_4/config.h:64]:
```suggestion
/* RGB matrix support. */
#define SPLIT_TRANSPORT_MIRROR
#define RGB_DISABLE_WHEN_USB_SUSPENDED
#define RGB_MATRIX_KEYPRESSES
#define RGB_MATRIX_FRAMEBUFFER_EFFECTS

// Startup values.
#define RGB_MATRIX_DEFAULT_HUE 0
#define RGB_MATRIX_DEFAULT_SAT 255
#define RGB_MATRIX_DEFAULT_VAL 64

// Rainbow swirl as startup mode.
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_LEFT_RIGHT

// Slow swirl at startup.
#define RGB_MATRIX_DEFAULT_SPD 32
```

File: keyboards/bastardkb/dilemma/4x6_4/info.json [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+    "caps_word": {
+        "enabled": true
+    },
```
Comment (drashna) [inline keyboards/bastardkb/dilemma/4x6_4/info.json:39]: Mostly a nit, this is easier to read/more compact: 
```suggestion
        "rgb_matrix": true,
        "caps_word": true
    },
```

**Comment by casuanoob** [comment]
Thanks for the review, @drashna! 
I'm a bit confused at the GitHub status - I've applied all the requested changes verbatim, but the PR is still apparently flagged as "Changes requested", is that just how GitHub PRs work?

**Comment by drashna** [comment]
> Thanks for the review, @drashna! I'm a bit confused at the GitHub status - I've applied all the requested changes verbatim, but the PR is still apparently flagged as "Changes requested", is that just how GitHub PRs work?

Yup, that's just how github works.  Until the change request/review is dismissed or updated, it shows up as "change requested".

File: keyboards/bastardkb/dilemma/4x6_4/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  // ├──────────────────────────────────────────────────────┤ ├──────────────────────────────────────────────────────┤
+       XXXXXXX, _______, DRGSCRL, SNIPING, EE_CLR,  QK_BOOT,    QK_BOOT, EE_CLR,  SNIPING, DRGSCRL, _______, XXXXXXX,
+  // ╰──────────────────────────────────────────────────────┤ ├──────────────────────────────────────────────────────╯
+                         XXXXXXX, KC_BTN2, KC_BTN1, KC_BTN3,    KC_BTN3, KC_BTN1, KC_BTN2, XXXXXXX
+  //                    ╰───────────────────────────────────╯ ╰───────────────────────────────────╯
+  ),
+};
+// clang-format on
+
+#ifdef POINTING_DEVICE_ENABLE
```
Comment (drashna) [inline keyboards/bastardkb/dilemma/4x6_4/keymaps/via/keymap.c:102]: Would rather see this using the core code, but isn't a blocker (more of a nitpick).

**Review by drashna — APPROVED** [review]

File: keyboards/bastardkb/dilemma/readme.md [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
 
-    make bastardkb/dilemma/3x5_2/splinky:default
-    make bastardkb/dilemma/3x5_2/assembled:default
+    make bastardkb/dilemma/3x5_3:via
+    make bastardkb/dilemma/4x6_4:via
+
+Flashing example for this keyboard:
+
+    make bastardkb/dilemma/3x5_3:via:flash
+    make bastardkb/dilemma/4x6_4:via:flash
```
Comment (fauxpark) [inline keyboards/bastardkb/dilemma/readme.md:19]:
```suggestion
    make bastardkb/dilemma/3x5_3:default
    make bastardkb/dilemma/4x6_4:default

Flashing example for this keyboard:

    make bastardkb/dilemma/3x5_3:default:flash
    make bastardkb/dilemma/4x6_4:default:flash
```

File: keyboards/bastardkb/dilemma/4x6_4/keymaps/default/readme.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Dilemma Max (4x6+4) default keymap
+
+> :bulb: Have a look at the [`via` keymap](../via) for a more feature-rich layout.
```
Comment (fauxpark) [inline keyboards/bastardkb/dilemma/4x6_4/keymaps/default/readme.md:3]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**